src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/court/CourtEventProcessor.kt

```diff
@@ -53,15 +53,13 @@ class CourtEventProcessor(
     }
     val commonPlatformHearingEvent = objectMapper.readValue<CommonPlatformHearingEvent>(commonPlatformHearing)
 
-    val uniqueDefendants = commonPlatformHearingEvent.hearing.prosecutionCases
+    val uniquePersonDefendants = commonPlatformHearingEvent.hearing.prosecutionCases
       .flatMap { it.defendants }
       .filterNot { it.isYouth }
+      .filter { it.isPerson() }
       .distinctBy { it.id }
-    val defendantIDs = uniqueDefendants.joinToString(" ") { it.id.toString() }
-    log.debug("Processing Common Platform Event with ${uniqueDefendants.size} distinct defendants with defendantId $defendantIDs")
 
-    val personDefendants = uniqueDefendants.filter { it.isPerson() }
-    personDefendants.forEach { defendant ->
+    uniquePersonDefendants.forEach { defendant ->
```

Comment:
could even do away with this variable and put the .foreach on the end of the above
:-) there is a lot to go at here.
For example, the `CommonPlatformHearingEvent` should actually just be a list of prosecutionCases as far as I can see